### PR TITLE
Change scheduler & input tensor shape

### PR DESCRIPTION
**Pull request by WoosukKwon**
This PR updates the scheduler and model code to use 2D tensors instead of 1D tensors. The change will enable using a wider range of libraries and hardware, and facilitate future optimizations like CUDA graph.

### CODE REVIEW

File: vllm/model_executor/input_metadata.py [hunk 1]
```diff
@@ -39,28 +39,28 @@ def __init__(
         self.max_context_len = max_context_len
         self.block_tables = block_tables
 
+        self.max_prompt_len = max(prompt_lens) if prompt_lens else 0
         self.to_cache = None
         if sliding_window is not None:
             # We need to keep the positions of sliding windows within
             # the key / value tables, this is helpful to know which
-            # elements we need to cache and where
+            # elements we need to cache and where.
```
Comment (zhuohan123) [inline vllm/model_executor/input_metadata.py:47]:
```suggestion
            # elements we need to cache.
```

**Review by zhuohan123 — APPROVED** [review]
LGTM! A small change may be needed in the comment.

File: vllm/model_executor/input_metadata.py (hunk 1 above)
Comment (WoosukKwon) [reply vllm/model_executor/input_metadata.py]: Fixed!

**Comment by yunfeng-scale** [comment]
i'm worried about performance degradation of these paddings, can you try benchmarking?

**Comment by frankxyy** [comment]
Hi, does the changing from 2D tensor to 1D tensor have some bad effects on the prefilling throughput? As some sequences with different lengths will be padded to the max length before calculation. Thus the total FLOPs needed is enhanced.

**Pull request merged**